wallet/wallet.go

```diff
@@ -1169,6 +1169,20 @@ out:
 	for {
 		select {
 		case txr := <-w.createTxRequests:
+			// If we're running in watch-only mode, there is no
+			// unlock possible. Therefore, we also don't need to
+			// prevent any unlocks from happening when doing coin
+			// selection in that mode.
+			if w.Manager.WatchOnly() {
+				tx, err := w.txToOutputs(
+					txr.outputs, txr.keyScope, txr.account,
+					txr.minconf, txr.feeSatPerKB,
+					txr.coinSelectionStrategy, txr.dryRun,
+				)
+				txr.resp <- createTxResponse{tx, err}
+				continue
+			}
```

Comment:
Seems can consolidate the code a bit here w/ something like: 
```go
var release func()
if !w.Manager.WatchOnly() {
    heldUnlock, err := w.holdUnlock()
    .... 
    release := heldUnlock.release()
}

tx, err := w.txToOutputs(
	txr.outputs, txr.keyScope, txr.account,
	txr.minconf, txr.feeSatPerKB,
	txr.coinSelectionStrategy, txr.dryRun,
)

if release != nil {
    release()
}

txr.resp <- createTxResponse{tx, err}
```

Thoughts?

Reply:
Great idea, done!